openmdao/core/driver.py

```diff
@@ -398,19 +398,20 @@ def set_desvar(self, name, value):
         val : ndarray or float
             value to set the parameter
         """
-        if self.root.unknowns._dat[name].val.size == 0:
+        val = self.root.unknowns._dat[name].val
+        if not isinstance(val, _ByObjWrapper) and \
+                       self.root.unknowns._dat[name].val.size == 0:
```

Comment:
yeah, we have a story in for that.  This was just getting his PR in, which makes things work if the driver happens to be gradient free and it has pbo desvars.